shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -378,6 +445,31 @@ - (void)setupNotificationCenterObservers {
                object:nil];
 }
 
+- (void)setupKeyboardAnimationCurve {
+  _keyboardAnimationStops.reset([[NSMutableArray alloc] init]);
+
+  // Keyboard animation spring curve for iOS 16 and below
+  _keyboardSpringCurve.reset([[KeyboardSpringCurve alloc] initWithStiffness:1000
+                                                               dampingRatio:1
+                                                                       mass:3
+                                                            initialVelocity:0
+                                                           settlingDuration:0.5]);
+  double settlingDuration = [[self keyboardSpringCurve] settlingDuration];
+  double displayRefreshRate = [DisplayLinkManager displayRefreshRate];
+
+  for (double time = 1 / displayRefreshRate; time < settlingDuration * 2;
+       time += 1 / displayRefreshRate) {
+    NSNumber* keyboardAnimationStop =
+        [NSNumber numberWithDouble:[[self keyboardSpringCurve] curveFunc:time]];
+    if (time <= settlingDuration + 1 / displayRefreshRate) {
+      [[self keyboardAnimationStops] addObject:keyboardAnimationStop];
+    } else {
+      [[self keyboardAnimationStops] addObject:[NSNumber numberWithDouble:1]];
+      break;
+    }
+  }
+}
```

Comment:
1. Why using the condition `time < settlingDuration * 2` ? 
Although I know you want reach ` [[self keyboardAnimationStops] addObject:[NSNumber numberWithDouble:1]];` and break out of loop at the end. But this condition will make people confuse......
Maybe using a while loop?
Maybe Like:
```objc
while(time <= settlingDuration + 1 / displayRefreshRate) {
      xxxx
      time += xxxxx
} 
[[self keyboardAnimationStops] addObject:[NSNumber numberWithDouble:1]];
```

2. Maybe better to cache `1 / displayRefreshRate` result, avoid to calculate it every time when using it.
Maybe called `double frameDuration = 1 / displayRefreshRate`.

Reply:
1. Yes confusing on the times 2. I'm not totally sure why it's necessary, but without it, the math works out the stops too quickly. By having it calculate the stops for double the duration, there's more granularity between each stop (especially the initial 25%). The additional stops after the half way point (60 for 120hz display) are all >99% so we aren't losing anything on the backend.

I like the while loop change :)

2. Changed

Reply:
The linter didn't like using the floating point `time` as an incrementer, so changing back to a for loop with an int counter